src/Sylius/Bundle/ChannelBundle/Attribute/AsChannelContextRequestResolver.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ChannelBundle\Attribute;
+
+#[\Attribute(\Attribute::TARGET_CLASS)]
+final class AsChannelContextRequestResolver
+{
+    public const SERVICE_TAG = 'sylius.context.channel.request_based.resolver';
```

Comment:
Completely besides the point, but this tag name is awful, we should think of changing it to something more concise